package/package.json

```diff
@@ -33,7 +33,8 @@
   "readmeFilename": "README.md",
   "dependencies": {
     "@types/react": "*",
-    "@types/react-native": "*"
+    "@types/react-native": "*",
+    "typescript": "^4.4.3"
```

Comment:
This should be a devDependency. It will otherwise bring in this version of TS to all user projects consuming react-native-xaml.

Reply:
Same for any of the `@types` bits.

Reply:
the package requires rebuilding its tsx/c++ files in response to projecting custom controls, which requires the types and tsc bits

Reply:
Is this something that is run by users consuming react-native-xaml, or on this repo before publish?

Reply:
users of react-native-xaml that need to project custom components

Reply:
Not going to dig too deeply in here, since this PR is meant to be for the JSValue change, but it might be worth talking to @acoates-ms a bit about what you're doing here with dependencies/TS generation.

I might be missing another usage, but the TSC usage I can find is an npm script internal to the pacakge, which are usually not expected to be run by users. There are some other build mechanisms that can be used though. Hoisting a wildcard version of the RN typings might also cause some unexpected behavior in some repos.